tests/decoding.c

```diff
@@ -120,6 +123,59 @@ static struct cmt *generate_encoder_test_data()
     return cmt;
 }
 
+static struct cmt *generate_encoder_test_basic_data()
```

Comment:
for general tests and considering future use cases, would be useful to have real protobuf payloads received from Prometheus and stored locally, as an example consider the text data files used for Prometheus text format:

https://github.com/fluent/cmetrics/tree/master/tests/data

as you can realize, most of them are associated with problems found decoding specific payloads, so in the future if we get any issue decoding an incoming Prometheus remote write payload would be easy to add a unit test to avoid regressions. I suggest to start with that from now, so maintenance gets easier later.

Reply:
Hmm, it should be in the first release. I'll handle on it.

Reply:
I used the actual payload of prometheus remote write from prometheus itself. It is still not using metadata but all of the ingested data from the server can handle with these patches.